tests/libraries/token/test_stake_token.py

```diff
@@ -0,0 +1,325 @@
+    with raises(Exception, message="market can't call trust buy unless market state is round 1 or round 2"):
+        mockMarket.callStakeTokenTrustedBuy(stakeToken, tester.a1, 1)
+
+    # call stake token successfully
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].ROUND1_REPORTING()) 
+    assert mockMarket.callStakeTokenTrustedBuy(stakeToken.address, tester.a1, 1)
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].ROUND2_REPORTING()) 
+    assert mockMarket.callStakeTokenTrustedBuy(stakeToken.address, tester.a1, 1)
+
+    mockMarket.setReportingState(tokenFixture.contracts['Constants'].LAST_DISPUTE()) 
+    with raises(Exception, message="market can't call trust buy unless market state is round 1 or round 2"):
+        mockMarket.callStakeTokenTrustedBuy(stakeToken, tester.a1, 1)
+
+@fixture(scope="session")
+def localSnapshot(fixture, kitchenSinkSnapshot):
+    fixture.resetToSnapshot(kitchenSinkSnapshot)
+    fixture.uploadAndAddToController('solidity_test_helpers/MockMarket.sol')
+    fixture.uploadAndAddToController('solidity_test_helpers/MockReportingWindow.sol')
+    fixture.uploadAndAddToController('solidity_test_helpers/MockUniverse.sol')
+    fixture.uploadAndAddToController('solidity_test_helpers/MockReputationToken.sol')
+    assert fixture.contracts['MockReputationToken']
+    universe = ABIContract(fixture.chain, kitchenSinkSnapshot['universe'].translator, kitchenSinkSnapshot['universe'].address)
+    market = ABIContract(fixture.chain, kitchenSinkSnapshot['binaryMarket'].translator, kitchenSinkSnapshot['binaryMarket'].address)
+    return initializeReportingFixture(fixture, universe, market)
+
+@fixture
+def universe(fixture, kitchenSinkSnapshot):
+    universe = ABIContract(fixture.chain, kitchenSinkSnapshot['universe'].translator, kitchenSinkSnapshot['universe'].address)
+    return universe
+
+@fixture
+def binaryMarket(fixture, kitchenSinkSnapshot):
+    market = ABIContract(fixture.chain, kitchenSinkSnapshot['binaryMarket'].translator, kitchenSinkSnapshot['binaryMarket'].address)
+    return market
+
+@fixture
+def cash(fixture, kitchenSinkSnapshot):
+    cash = ABIContract(fixture.chain, kitchenSinkSnapshot['cash'].translator, kitchenSinkSnapshot['cash'].address)
+    return cash
+
+@fixture
+def categoricalMarket(fixture, kitchenSinkSnapshot):
+    market = ABIContract(fixture.chain, kitchenSinkSnapshot['categoricalMarket'].translator, kitchenSinkSnapshot['categoricalMarket'].address)
+    return market
+
+@fixture
+def tokenFixture(fixture, universe, localSnapshot):
+    fixture.resetToSnapshot(localSnapshot)
+    value = 1 * 10**6 * 10**18
+    reputationToken = fixture.applySignature('ReputationToken', universe.getReputationToken())
+    # Give tester reputation tokens
+    for testAccount in [tester.a4, tester.a5, tester.a6, tester.a7, tester.a8, tester.a9]:
+        reputationToken.transfer(testAccount, value)
```

Comment:
This should be moved to `localSnapshot` rather than here.  By having it here, this code is re-run before every test which increases time required to run the tests since transaction execution is slow.  Glancing at this function, I think most/all of this logic could be moved to `localSnapshot` instead of here, which will also allow you to remove the `universe` dependency on this fixture which will make resolving the issue I mentioned above about ordering easier to address.

I consider `Controlled.sol` to be a good example of local fixture setup, you may want to check it out as a reference, or just ping me in #dev if none of this is making sense.  😄